### PR TITLE
Added developing measurement tutorial

### PR DESCRIPTION
**Pull request by marseel**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Tutorial about developing measurements in CL2.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
/assign @wojtek-t 
/assign @mm4tt 
**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: clusterloader2/docs/DEVELOPING_MEASUREMENT.md [hunk 1]
```diff
@@ -0,0 +1,579 @@
...
+		return y
+	}
+	return x
+}
+
+func (m *maxRunningPodsMeasurement) checkPod(oldObj, newObj interface{}) {
+	isDeleted := newObj == nil
+	isNewObject := oldObj == nil
+
+	newPod, _ := newObj.(*corev1.Pod)
```
Comment (wojtek-t) [inline clusterloader2/docs/DEVELOPING_MEASUREMENT.md:150]: silently ignoring an error is an anti-pattern :)

Comment (wojtek-t) [reply clusterloader2/docs/DEVELOPING_MEASUREMENT.md:150]: BTW - you're not handling tombstones - e.g.:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replicaset/replica_set.go#L333

File: clusterloader2/docs/DEVELOPING_MEASUREMENT.md [hunk 2]
```diff
@@ -0,0 +1,579 @@
...
+	oldPod, _ := oldObj.(*corev1.Pod)
+
+	defer func() {
+		m.maxRunningPods = MaxInt(m.maxRunningPods, m.currentRunningPods)
+		klog.V(2).Infof("Max: %d, current: %d", m.maxRunningPods, m.currentRunningPods)
+	}()
+
+	if isDeleted {
```
Comment (wojtek-t) [inline clusterloader2/docs/DEVELOPING_MEASUREMENT.md:158]: I think we can make it much simpler by doing:
```
func isPodRunning(obj interface{}) bool {
  pod := // convert interface to pod
  return pod != nil && pod.Status.Phase == corev1.PodRunning
}

change := 0
if isPodRunning(oldObj) {
  change--
}
if isPodRunning(newObj) {
  change++
}
```

File: clusterloader2/docs/DEVELOPING_MEASUREMENT.md [hunk 3]
```diff
@@ -0,0 +1,579 @@
...
+
+```golang
+func MaxInt(x, y int) int {
+	if x < y {
+		return y
+	}
+	return x
+}
+
+func (m *maxRunningPodsMeasurement) checkPod(oldObj, newObj interface{}) {
```
Comment (wojtek-t) [inline clusterloader2/docs/DEVELOPING_MEASUREMENT.md:146]: Discuss why we don't need locking?

[or rather why we're ignoring it in all measurements...]

File: clusterloader2/docs/DEVELOPING_MEASUREMENT.md [hunk 4]
```diff
@@ -0,0 +1,579 @@
...
+  selector:
+    matchLabels:
+      k8s-app: node-local-dns
+  namespaceSelector:
+    matchNames:
+      - kube-system
+```
+Any monitor needs to be created within monitoring namespace.
+Most important is specifying interval how often metric will be scraped (here once every 10 minutes).
+The more pods you want to scrape the smaller interval should be and/or Prometheus server should have more resources.
```
Comment (wojtek-t) [inline clusterloader2/docs/DEVELOPING_MEASUREMENT.md:415]: s/smaller/larger/

[or smaller frequency]

File: clusterloader2/docs/DEVELOPING_MEASUREMENT.md (hunk 2 above)
Comment (marseel) [reply clusterloader2/docs/DEVELOPING_MEASUREMENT.md:158]: Good idea, done.

File: clusterloader2/docs/DEVELOPING_MEASUREMENT.md (hunk 4 above)
Comment (marseel) [reply clusterloader2/docs/DEVELOPING_MEASUREMENT.md:415]: Done.

File: clusterloader2/docs/DEVELOPING_MEASUREMENT.md (hunk 3 above)
Comment (marseel) [reply clusterloader2/docs/DEVELOPING_MEASUREMENT.md:147]: Added lock

File: clusterloader2/docs/DEVELOPING_MEASUREMENT.md (hunk 1 above)
Comment (marseel) [reply clusterloader2/docs/DEVELOPING_MEASUREMENT.md]: Handled error, but I've decided to skip handling tombstone and treat it as not running pod.

**Pull request merged**